src/components/modal/index.tsx

```diff
@@ -32,6 +36,32 @@ export interface ModalProps {
 }
 
 export const Modal: React.FC<ModalProps> = ({ title, show, onExit, children }) => {
+    const ref = useRef<HTMLDivElement>(null);
+    // Close modal on Esc press
+    const handleEscKey = useCallback((event: KeyboardEvent) => {
+        if (event.key === KEY_NAME_ESC) {
+            onExit();
+        }
+    }, [onExit]);
+
+    // Close on outside click
+    const handleClick = useCallback((event: MouseEvent) => {
+        // @ts-ignore EventTarget is not mapping on Node but works just fine
+        if (ref?.current?.contains && !ref.current.contains(event.target)) {
+            onExit();
+        }
+    }, [onExit, ref]);
+
+    useEffect(() => {
+        document.addEventListener(KEY_EVENT_TYPE, handleEscKey, false);
+        document.addEventListener(MOUSE_UP, handleClick);
+
+        return () => {
+            document.removeEventListener(KEY_EVENT_TYPE, handleEscKey, false);
+            document.removeEventListener(MOUSE_UP, handleClick);
+        };
+    }, [handleClick, handleEscKey]);
```

Comment:
The constant terms are honestly not necessary here. Let's just write the string values directly.

Reply:
I'd disagree. It's being used more than once, so I'd leave as is, even if they are fixed names.

Reply:
Well `KEY_NAME_ESC` is only being used once. Sure, the event names are used twice, but they are a few lines apart and are already defined in the TypeScript global event handler mapping.